packages/block-library/src/avatar/edit.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	InspectorControls,
+	useBlockProps,
+	__experimentalGetSpacingClassesAndStyles as useSpacingProps,
+} from '@wordpress/block-editor';
+import { PanelBody, ResizableBox, RangeControl } from '@wordpress/components';
+import { __, isRTL } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useUserAvatar, useCommentAvatar } from './utils';
+
+const renderInspectorControls = ( { setAttributes, avatar, attributes } ) => (
+	<InspectorControls>
+		<PanelBody title={ __( 'Avatar Settings' ) }>
+			<RangeControl
+				label={ __( 'Image size' ) }
+				onChange={ ( newWidth ) =>
+					setAttributes( {
+						width: newWidth,
+						height: newWidth,
+					} )
+				}
+				min={ avatar.minSize }
+				max={ avatar.maxsize }
+				initialPosition={ attributes?.width }
+				value={ attributes?.width }
+			/>
+		</PanelBody>
+	</InspectorControls>
+);
+
+const renderAvatar = ( {
+	setAttributes,
+	attributes,
+	avatar,
+	blockProps,
+	isSelected,
+} ) => (
+	<ResizableBox
+		size={ {
+			width: attributes.width,
+			height: attributes.height,
+		} }
+		showHandle={ isSelected }
+		onResizeStop={ ( event, direction, elt, delta ) => {
+			setAttributes( {
+				height: parseInt( attributes.height + delta.height, 10 ),
+				width: parseInt( attributes.width + delta.width, 10 ),
+			} );
+		} }
+		lockAspectRatio
+		enable={ {
+			top: false,
+			right: ! isRTL(),
+			bottom: true,
+			left: isRTL(),
+		} }
+		minWidth={ avatar.minSize }
+		maxWidth={ avatar.maxSize }
+	>
+		<img src={ avatar.src } alt={ avatar.alt } { ...blockProps } />
+	</ResizableBox>
+);
+
+export default function Edit( {
+	attributes,
+	context: { commentId, postId, postType },
+	setAttributes,
+	isSelected,
+} ) {
+	const blockProps = useBlockProps();
+	const spacingProps = useSpacingProps( attributes );
+
+	// I would like to not call useCommentAvatar API requests if there is no commentId
+	// but is not recommended having conditional hook. Any ideas?
+	const commentAvatar = useCommentAvatar( { commentId } );
+	const userAvatar = useUserAvatar( { postId, postType } );
```

Comment:
Is there a better way to achieve this (without having to request if we have no comment Id) 😅

Reply:
Hmm, in this case I think you might have create an `UserEdit` and `CommentEdit` components and the current `Edit` component be a wrapper for those two. I don't think there is any other way. So, something like:

```jsx
export default function Edit( {
	context: { commentId, postId, postType },
	...props
} ) {

  if (commentId) {
    return <CommentEdit {...props}/>;
  } else {
    return <UserEdit postId={postId} postType={postType} {...props} />;
  }
}
```

Then, in the `CommentEdit` component, you would not have to call the `useCommentAvatar()`.

Reply:
Yes, that’s the unfortunate consequence of React hooks. If you want branch your code you need to be creative 😅

Reply:
Done!